src/utils/messaging-outbox/src/services/outbox_transactional_processor.rs

```diff
@@ -288,13 +292,23 @@ impl ProducerRelayJob {
         relay_routes_static_info: Arc<RoutesStaticInfo>,
         producer_name: String,
         consumer_names: Vec<String>,
+        metrics: Arc<OutboxTransactionalProcessorMetrics>,
     ) -> Self {
+        // Initialize labeled metrics so they show up in the output early
+        for consumer_name in &consumer_names {
+            metrics
+                .failed_consumers
+                .with_label_values(&[&producer_name, consumer_name])
+                .set(0);
+        }
```

Comment:
This is coupling the service to metrics library directly. 

I think we should have a higher level API to cover it. 
I.e.:
    `metrics.track_routing_path(&producer_name, consumer_name)`
and then
    `metrics.routing_failed(&producer_name, consumer_name)`.
    
The metrics itself should be a trait, not a struct, so that we can unit test this applying mocking or dummy implementation.